napari/layers/utils/layer_utils.py

```diff
@@ -204,71 +214,82 @@ def _nanmax(array):
     return max_value
 
 
-def calc_data_range(data, rgb=False) -> Tuple[float, float]:
+def calc_data_range(data, rgb: bool = False) -> None | Tuple[float, float]:
     """Calculate range of data values. If all values are equal return [0, 1].
 
     Parameters
     ----------
     data : array
         Data to calculate range of values over.
     rgb : bool
-        Flag if data is rgb.
+        Flag if data is rgb. If so, return [0, 255].
 
     Returns
     -------
-    values : pair of floats
-        Minimum and maximum values in that order.
+    values : None | Tuple[float, float]
+        Minimum and maximum values in that order or None if dtype is float.
 
     Notes
     -----
-    If the data type is uint8, no calculation is performed, and 0-255 is
+    If the data type is uint8 or rgb, no calculation is performed, and 0-255 is
     returned.
     """
-    if data.dtype == np.uint8:
-        return (0, 255)
-
-    center: Union[int, List[int]]
-
-    if data.size > 1e7 and (data.ndim == 1 or (rgb and data.ndim == 2)):
-        # If data is very large take the average of start, middle and end.
-        center = int(data.shape[0] // 2)
-        slices = [
-            slice(0, 4096),
-            slice(center - 2048, center + 2048),
-            slice(-4096, None),
-        ]
+    shape = data.shape
+    chunk_size = (
+        _get_chunk_size(data) if not isinstance(data, np.ndarray) else None
+    )
+
+    dtype = normalize_dtype(getattr(data, 'dtype', None))
+    if dtype == np.uint8 or (rgb and dtype == np.uint8):
+        return 0, 255
```

Comment:
This is equivalent AFAIK:

```suggestion
    if dtype == np.uint8:
        return 0, 255
```

Reply:
I changed this to or rgb. Think is here we can still get `np.uint16` rgb for example. Since Vispy casts this to `np.uint8` this is now valid I believe.